### PR TITLE
Adding a sampler for the krb5kdc binary.

### PR DESCRIPTION
**Pull request by kylebartush-twitter**
Provides telemetry to track MIT kerberos ticket requests served by the krb5kdc binary. This is accomplished by attaching user space probes to the following functions: finish_process_as_req, finish_dispatch_cache and process_tgs_req. Each function exports a call count that is broken down by the resulting error code. Since there is a very large list of possible error codes, the first 30 error codes are exported. All other error code values are exported as UNKNOWN.

Each error code is reformatted to better fit metric naming standards: "KRB5KDC_ERR_BAD_PVNO" -> "bad_pvno"

krb5kdc/finish_process_as_req/{ERROR_CODE} - count of finish_process_as_req calls by error
krb5kdc/finish_dispatch_cache/{ERROR_CODE} - count of finish_dispatch_cache calls by error
krb5kdc/process_tgs_req/{ERROR_CODE} - count of process_tgs_req calls by error

### CODE REVIEW

File: src/samplers/krb5kdc/stat.rs [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+    EnumString,
+    Eq,
+    IntoStaticStr,
+    PartialEq,
+    Hash,
+    Serialize,
+)]
+#[serde(deny_unknown_fields, try_from = "&str", into = "&str")]
+pub enum Krb5kdcStatistic {
+    #[strum(serialize = "krb5kdc/finish_process_as_req/unknown")]
```
Comment (brayniac) [inline src/samplers/krb5kdc/stat.rs:28]: Can you add some documentation to `docs/METRICS.md` for these new metrics?

File: src/samplers/krb5kdc/stat.rs [hunk 2]
```diff
@@ -0,0 +1,706 @@
...
+
+    #[strum(serialize = "krb5kdc/process_tgs_req/svc_unavailable")]
+    ProcessTgsReqSvcUnavailable,
+}
+
+impl Krb5kdcStatistic {
+    pub fn bpf_table(self) -> &'static str {
+        match self {
+            Self::FinishProcessAsReqUnknown => "counts_finish_process_as_req",
+
```
Comment (brayniac) [inline src/samplers/krb5kdc/stat.rs:312]: I'm not sure theres any readability benefit to these newlines in the match

File: src/samplers/krb5kdc/stat.rs [hunk 3]
```diff
@@ -0,0 +1,706 @@
...
+            Self::ProcessTgsReqPathNotAccepted => "counts_process_tgs_req",
+
+            Self::ProcessTgsReqSvcUnavailable => "counts_process_tgs_req",
+        }
+    }
+
+    pub fn bpf_entry(self) -> &'static str {
+        match self {
+            Self::FinishProcessAsReqUnknown => "UNKNOWN",
+
```
Comment (brayniac) [inline src/samplers/krb5kdc/stat.rs:502]: Same here

File: src/samplers/krb5kdc/stat.rs (hunk 1 above)
Comment (kylebartush-twitter) [reply src/samplers/krb5kdc/stat.rs:28]: Add a section on krb5kdc metrics.

File: src/samplers/krb5kdc/stat.rs (hunk 2 above)
Comment (kylebartush-twitter) [reply src/samplers/krb5kdc/stat.rs]: Cleaned up.

File: src/samplers/krb5kdc/stat.rs (hunk 3 above)
Comment (kylebartush-twitter) [reply src/samplers/krb5kdc/stat.rs]: Cleaned up.

**Review by brayniac — APPROVED** [review]
Looks good to me

**Pull request merged**